be/src/exec/blocking_join_node.h

```diff
@@ -84,11 +88,20 @@ class BlockingJoinNode : public ExecNode {
     // This should be the same size as the left child tuple row.
     int _result_tuple_row_size;
 
+    /// Row assembled from all lhs and rhs tuples used for evaluating the non-equi-join
+    /// conjuncts for semi joins. Semi joins only return the lhs or rhs output tuples,
+    /// so this tuple is temporarily assembled for evaluating the conjuncts.
+    TupleRow* _semi_join_staging_row;
```

Comment:
now we can't support no-equi-join,  will you add no-euqal-join semi join latter?

Reply:
No，This comment is not very accurate. I will revise this part.